src/main/java/edu/wpi/grip/core/Step.java

```diff
@@ -78,13 +78,25 @@ protected Pipeline getPipeline() {
         return this.pipeline;
     }
 
+    private synchronized void runPerformIfPossible() {
+        for (InputSocket<?> inputSocket : inputSockets) {
+            if (!inputSocket.getValue().isPresent()) {
+                for (OutputSocket<?> outputSocket : outputSockets) {
+                    outputSocket.resetValueToInitial();
+                }
+                return;
```

Comment:
This will eventually be replaced with the event posting but okay. I can take care of that.